app/src/main/java/org/stepik/android/presentation/course/CoursePresenter.kt

```diff
@@ -126,11 +126,11 @@ constructor(
      * Data initialization variants
      */
     fun onCourseId(courseId: Long, viewSource: CourseViewSource, forceUpdate: Boolean = false) {
-        observeCourseData(courseInteractor.getCourseHeaderData(courseId), viewSource, forceUpdate)
+        observeCourseData(courseInteractor.getCourseHeaderData(courseId, canUseCache = !forceUpdate), viewSource, forceUpdate)
     }
 
     fun onCourse(course: Course, viewSource: CourseViewSource, forceUpdate: Boolean = false) {
-        observeCourseData(courseInteractor.getCourseHeaderData(course), viewSource, forceUpdate)
+        observeCourseData(courseInteractor.getCourseHeaderData(course, canUseCache = !forceUpdate), viewSource, forceUpdate)
```

Comment:
On connection error it will fallback to an old `course` model instead of one from `state`

> При отсутствии интернета fallback будет на старую модель `course`, а не ту, которая в `state`